src/dxtbx/format/FormatNXmx.py

```diff
@@ -63,4 +64,16 @@ def get_raw_data(self, index):
         nxmx = dxtbx.nexus.nxmx.NXmx(self._cached_file_handle)
         nxdata = nxmx.entries[0].data[0]
         nxdetector = nxmx.entries[0].instruments[0].detectors[0]
-        return dxtbx.nexus.get_raw_data(nxdata, nxdetector, index)
+        raw_data = dxtbx.nexus.get_raw_data(nxdata, nxdetector, index)
+        if self._bit_depth_readout:
+            # if 32 bit then it is a signed int, I think if 8, 16 then it is
+            # unsigned with the highest two values assigned as masking values
+            if self._bit_depth_readout == 32:
+                top = 2 ** 31
+            else:
+                top = 2 ** self._bit_depth_readout
+            for data in raw_data:
+                d1d = data.as_1d()
+                d1d.set_selected(d1d == top - 1, -1)
+                d1d.set_selected(d1d == top - 2, -2)
+        return raw_data
```

Comment:
Hi @rjgildea, I found this code while working on some NeXus stuff.  Is this Eiger specific?  Would it apply to other NXmx data?  Thanks!